experimental/packages/exporter-logs-otlp-proto/test/node/OTLPLogsExporter.test.ts

```diff
@@ -0,0 +1,416 @@
+        const mockRes = new MockedResponse(200);
+        cb(mockRes);
+        mockRes.send('success');
+        return fakeRequest as any;
+      });
+    });
+
+    it('should log the error message', done => {
+      collectorExporter.export(logs, result => {
+        assert.strictEqual(result.code, ExportResultCode.FAILED);
+        // @ts-expect-error verify error code
+        assert.strictEqual(result.error.code, 400);
+        done();
+      });
+
+      sinon.stub(http, 'request').callsFake((options: any, cb: any) => {
+        const mockResError = new MockedResponse(400);
+        cb(mockResError);
+        mockResError.send('failed');
+
+        return fakeRequest as any;
+      });
+    });
+  });
+  describe('export - with compression', () => {
+    beforeEach(() => {
+      collectorExporterConfig = {
+        headers: {
+          foo: 'bar',
+        },
+        hostname: 'foo',
+        url: 'http://foo.bar.com',
+        keepAlive: true,
+        compression: CompressionAlgorithm.GZIP,
+        httpAgentOptions: { keepAliveMsecs: 2000 },
+      };
+      collectorExporter = new OTLPLogsExporter(collectorExporterConfig);
+      logs = [];
+      logs.push(Object.assign({}, mockedReadableLogRecord));
+    });
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should successfully send the logs', done => {
+      const fakeRequest = new Stream.PassThrough();
+      sinon.stub(http, 'request').returns(fakeRequest as any);
+      const spySetHeader = sinon.spy();
+      (fakeRequest as any).setHeader = spySetHeader;
+
+      let buff = Buffer.from('');
+      fakeRequest.on('end', () => {
+        const unzippedBuff = zlib.gunzipSync(buff);
+        const ExportLogsServiceRequestProto = getExportRequestProto(
+          ServiceClientType.LOGS
+        );
+        const data = ExportLogsServiceRequestProto.decode(unzippedBuff);
+        const json = data?.toJSON() as IExportLogsServiceRequest;
+        const log1 = json.resourceLogs?.[0].scopeLogs?.[0].logRecords?.[0];
+        assert.ok(typeof log1 !== 'undefined', "log doesn't exist");
+        ensureExportedLogRecordIsCorrect(log1);
+
+        ensureExportLogsServiceRequestIsSet(json);
+        assert.ok(spySetHeader.calledWith('Content-Encoding', 'gzip'));
+
+        done();
+      });
+
+      fakeRequest.on('data', chunk => {
+        buff = Buffer.concat([buff, chunk]);
+      });
+
+      const clock = sinon.useFakeTimers();
+      collectorExporter.export(logs, () => {});
+      clock.tick(200);
+      clock.restore();
+    });
+  });
+});
+
+describe('export - real http request destroyed before response received', () => {
+  let collectorExporter: OTLPLogsExporter;
+  let collectorExporterConfig: OTLPExporterNodeConfigBase;
+  let logs: ReadableLogRecord[];
+  const server = http.createServer((_, res) => {
+    setTimeout(() => {
+      res.statusCode = 200;
+      res.end();
+    }, 200);
+  });
+  before(done => {
+    server.listen(8080, done);
```

Comment:
```suggestion
    server.listen(8082, done);
```
this will interfere with the existing trace exporter tests when run in parallel. In metrics we use `8081` so I think we could use `8082` here.